core/include/detray/navigation/navigator.hpp

```diff
@@ -726,9 +725,18 @@ class navigator {
             navigation.m_heartbeat &= init(track, navigation, cfg);
 
             // Sanity check: Should never be the case after complete update call
-            if (navigation.trust_level() != navigation::trust_level::e_full ||
-                navigation.is_exhausted()) {
-                navigation.abort();
+            if (navigation.trust_level() != navigation::trust_level::e_full) {
+                // Try to save the navigation flow: Look further behind the
+                // track
+                auto loose_cfg{cfg};
+                loose_cfg.overstep_tolerance = -10.f * cfg.max_mask_tolerance;
```

Comment:
The mask tolerance is in mm, the overstepping tolerance is in um, and the maximal mask tolerance should always be a large value (otherwise it makes no sense to have it). I mainly did this for cases in the telescope detector where surfaces stick out of the portals around them due to the mask tolerance. If the track then steps onto such a surface in the tolerance band of the mask, it can end up outside of the volume. It is then unable to find any surface of the volume behind it, because the mask tolerance on a surface that is some distance away from the previous track position can be much bigger that the overstepping tolerance. So I was thinking that I want this ```loose_cfg``` to be able to adapt to that situation. But I think adding a min value here might be a good idea in case someone sets a super small max mask tolerance

Reply:
Done